torch/_guards.py

```diff
@@ -797,6 +801,9 @@ def is_dict_key(self):
         # Recurse until you either hit a ConstDictKey or a Source
         return self.base.is_dict_key()
 
+    def is_ephemeral(self):
+        return self.base.is_ephemeral()
```

Comment:
is it even possible to chain an ephemeral source into a larger source 🤔

Reply:
The answer is apparently yes, because ephemeral sources identify *tensors*, so you chain size/stride accessors off of them.